File: Cap 05/Topico 5.4/Topico 5.4.1/Pos-Teste Fluxo Falso/Interativo/repitaINT.go
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 5 - Topico 5.4.1
 * 
 * Programa ..: LACO CONDICIONAL POS-TESTE COM FLUXO FALSO INTERATIVO
 */

package main

import (
  "fmt"
  "strings"
)

var (
  N    int
  R    int
  RESP string
)
func main() {
  fmt.Println("LACO POS-TESTE COM FLUXO FALSO")
  fmt.Println("Multiplicacao de numero por 3 por N vezes")
  fmt.Println()
  RESP = "S"
  for { // simulacao "repita"
    fmt.Print("Entre valor numerico inteiro: ")
    fmt.Scanln(&N)
    R = N * 3
    fmt.Println("O numero informado X 3 =", R)
    fmt.Println()
    fmt.Print("Deseja continuar? (S/N): ")
    fmt.Scanln(&RESP)
    fmt.Println()
    if (strings.ToUpper(RESP) != "S") {break} // simulacao "ate_que (RESP <> "S")"   
  }
}

File: Cap 05/Topico 5.4/Topico 5.4.1/Pos-Teste Fluxo Falso/Interativo/repitaINT.cpp
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 5 - Topico 5.4.1
 * 
 * Programa ..: LACO CONDICIONAL POS-TESTE COM FLUXO FALSO INTERATIVO
 * 
 */

#include <iostream>
#include <cctype>
using namespace std;

int N, R;
char RESP;
int main(void) {
  cout << "LACO POS-TESTE COM FLUXO FALSO" << endl;
  cout << "Multiplicacao de numero por 3 por N vezes" << endl << endl;
  RESP = 'S';
  do { // simulacao laco "repita"
    cout << "Entre valor numerico inteiro: ";
    cin >> N;
    R = N * 3;
    cout << "O numero informado X 3 = " << R << endl << endl;
    cout << "Deseja continuar? (S/N): ";
    cin >> RESP;
    RESP = toupper(RESP);
    cout << endl;
  } while (not (RESP != 'S')); // simulacao "ate_que (RESP <> "S")"
  return 0;
}
